api/all/src/main/java/io/opentelemetry/api/GlobalOpenTelemetry.java

```diff
@@ -115,4 +122,25 @@ public static Tracer getTracer(String instrumentationName, String instrumentatio
   public static ContextPropagators getPropagators() {
     return get().getPropagators();
   }
+
+  @Nullable
+  private static OpenTelemetry maybeAutoConfigure() {
+    final Class<?> openTelemetrySdkAutoConfiguration;
+    try {
+      openTelemetrySdkAutoConfiguration =
+          Class.forName("io.opentelemetry.sdk.autoconfigure.OpenTelemetrySdkAutoConfiguration");
+    } catch (ClassNotFoundException e) {
+      return null;
+    }
+
+    try {
+      Method initialize = openTelemetrySdkAutoConfiguration.getMethod("initialize");
+      return (OpenTelemetry) initialize.invoke(null);
+    } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+      throw new IllegalStateException(
+          "OpenTelemetrySdkAutoConfiguration detected on classpath "
+              + "but could not invoke initialize method. This is a bug in OpenTelemetry.",
```

Comment:
this could also be due to a restrictive security manager

Reply:
For a public method, as far as I know there wouldn't be any situation that could be blocked by security manager, is there one?

Reply:
I thought security managers could block all reflection, but I wouldn't swear to it.

Reply:
Yes, a security manager can block all reflection, even for public methods, and even just calls to "getMethod", although I would suspect that's extremely rare these days.

Reply:
Had no idea - do you think I should change the message? Wondering how realistic it is to disable public reflection vs adding noise to this code.

Reply:
Maybe just "This could be a bug in OpenTelemetry" ?  Really, it doesn't matter all that much.